app/Http/Controllers/CommentController.php

```diff
@@ -93,8 +158,14 @@ public function update(Request $request, $id)
      */
     public function destroy(\App\Comment $comment)
     {
+
+        if ( \Auth::user()->type != 'teacher'){
+            return redirect()->back()->with('error', 'Your are not allow to delete posts');
+        }
+
         $comment->delete();
         \App\Point::deAssign($comment->author()->id,\App\Point::BENEFACTOR_TYPE_COMMENTED );
         $comment->author()->deletePoints(\App\Point::BENEFACTOR_TYPE_COMMENTED, true);
+        return redirect()->back()->with('success', 'Your are allow to delete posts');
```

Comment:
Instead of saying they are allow to delete posts, we should inform them that the post has now been deleted.